code/modules/hallucinations/types/mirage.dm

```diff
@@ -0,0 +1,188 @@
+/datum/hallucination/mirage
+	max_power = 30
+	duration = 120
+	var/number = 3
+	var/list/mirages = list()
+
+/datum/hallucination/mirage/start()
+	duration = rand(100, 150)
+	var/list/possible_points = list()
+	for(var/turf/simulated/floor/F in view(holder, world.view+1))
+		possible_points += F
+	if(possible_points.len)
+		for(var/i = 1; i <= number; i++)
+			var/image/thing = generate_mirage()
+			mirages += thing
+			thing.loc = pick(possible_points)
+		holder.client.images += mirages
+
+/datum/hallucination/mirage/Destroy()
+	if(holder.client)
+		holder.client.images -= mirages
+	. = ..()
+
+/datum/hallucination/mirage/proc/generate_mirage()
+	var/icon/T = new('icons/obj/trash.dmi')
+	return image(T, pick(T.IconStates()), layer = OBJ_LAYER)
+
+/datum/hallucination/mirage/end()
+	if(holder.client)
+		holder.client.images -= mirages
+	..()
+
+
+/datum/hallucination/mirage/bleeding
+	min_power = 30
+	max_power = INFINITY
+	duration = 350
+	allow_duplicates = FALSE
+	number = 4
+	var/part = "chest"
+
+/datum/hallucination/mirage/bleeding/start()
+	number = min(round(holder.hallucination/10), 7)	//cap at 7 times for duration's sake
+	for(var/i = 1; i <= number; i++)
+		addtimer(CALLBACK(src, .proc/show_mirage), rand(30,50)*i)	//every 3 to 5 seconds
+	if(ishuman(holder))
+		var/mob/living/carbon/human/H = holder
+		part = LAZYPICK(H.organs, "chest")
+	to_chat(holder, SPAN_DANGER("The flesh on your [part] splits open. It doesn't hurt, but the blood won't stop coming..."))
+
+
+/datum/hallucination/mirage/bleeding/generate_mirage()
+	if(prob(min(holder.hallucination, 80)))
+		var/image/I = image('icons/effects/blood.dmi', pick("mgibbl1", "mgibbl2", "mgibbl3", "mgibbl4", "mgibbl5", "mfloor6", "mfloor7"), layer = TURF_LAYER)
+		I.color = holder.species.blood_color
```

Comment:
```suggestion
		I.color = holder.species?.blood_color
```
Species might be absent if it is a carbon.

Reply:
so THAT'S how those question marks are used.